src/axom/primal/geometry/Triangle.hpp

```diff
@@ -200,78 +201,106 @@ class Triangle
+      return numerics::determinant< double > (pA[0], pA[1], pA[2],
+                                              pB[0], pB[1], pB[2],
+                                              pC[0], pC[1], pC[2]);
       // clang-format on
     }
   }
 
 public:
   /*!
    * \brief Returns the barycentric coordinates of a point within a triangle
-   * \return The barycentric coordinates of the triangle inside a Point<T,3>
+   *
+   * \param [in] p The point at which we want to compute barycentric coordinates
+   * \param [in] skipNormalization Determines if the result should be normalized
+   * by the area of the triangle. One might want to skip this if they were
+   * only interested in the relative weights rather than their actual amounts
+   * 
    * \pre The point lies in this triangle's plane.
-   * \post The barycentric coordinates sum to 1.
-   * Adapted from Real Time Collision Detection by Christer Ericson.
+   * \post The barycentric coordinates sum to 1 when \a skipNormalization is false
+   * Otherwise, the sum of coordinates will be proportional to area of the triangle
+   *
+   * Algorithm adapted from Real Time Collision Detection by Christer Ericson.
    */
-  Point<double, 3> physToBarycentric(const PointType& p) const
+  Point<double, 3> physToBarycentric(const PointType& p,
+                                     bool skipNormalization = false) const
   {
-    using axom::numerics::determinant;
+    // Query point needs to be in triangle's plane
     SLIC_CHECK(axom::utilities::isNearlyEqual(ppedVolume(p), 0.));
 
     Point<double, 3> bary;
 
-    Vector<double, 3> u =
-      VectorType::cross_product(VectorType(m_points[0], m_points[1]),
-                                VectorType(m_points[0], m_points[2]));
-    const double x = std::abs(u[0]);
-    const double y = std::abs(u[1]);
-    const double z = std::abs(u[2]);
-
-    double ood = 1.0 / u[2];  // compute in xy plane by default
-    int c0 = 0;
-    int c1 = 1;
-
-    if(x >= y && x >= z)
-    {
-      // compute in yz plane
-      c0 = 1;
-      c1 = 2;
-      ood = 1.0 / u[0];
-    }
-    else if(y >= x && y >= z)
-    {
-      // compute in xz plane
-      c0 = 0;
-      c1 = 2;
-      ood = -1.0 / u[1];
-    }
+    auto triArea2D =
+      [](double x1, double y1, double x2, double y2, double x3, double y3)
+      -> double { return (x1 - x2) * (y2 - y3) - (x2 - x3) * (y1 - y2); };
 
     // References to triangle vertices for convenience
     const PointType& A = m_points[0];
     const PointType& B = m_points[1];
     const PointType& C = m_points[2];
 
-    // Compute ood * area of each sub-triangle
-    // clang-format off
-    bary[0] = ood * determinant(p[c0]-B[c0], p[c1]-B[c1],
-                                B[c0]-C[c0], B[c1]-C[c1]);
-    bary[1] = ood * determinant(p[c0]-C[c0], p[c1]-C[c1],
-                                C[c0]-A[c0], C[c1]-A[c1]);
-    bary[2] = 1. - bary[0] - bary[1];
-    // clang-format on
+    // unnormalized triangle normal
```

Comment:
denormalize?

Reply:
I think "unnormalized" is correct here since we're not normalizing it.